src/librustc_mir_build/build/expr/as_rvalue.rs

```diff
@@ -225,7 +225,11 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             }
             ExprKind::Assign { .. } | ExprKind::AssignOp { .. } => {
                 block = unpack!(this.stmt_expr(block, expr, None));
-                block.and(this.unit_rvalue())
+                block.and(Rvalue::Use(Operand::Constant(box Constant {
+                    span: expr_span,
+                    user_ty: None,
+                    literal: ty::Const::zero_sized(self.hir.tcx(), self.hir.tcx().types.unit),
```

Comment:
You need to write `this` instead of `self`, note how the old code used `this.unit_rvalue()`.

Reply:
I just used copy paste for now, seems like I missed this :laughing: